src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -58,6 +58,53 @@ public void GetProcessesByName_RemoteMachineNameUnix_ThrowsPlatformNotSupportedE
             Assert.Throws<PlatformNotSupportedException>(() => Process.GetProcessesByName(currentProcess.ProcessName, machineName));
         }
 
+        [Fact]
+        public void GetProcesses_LongProcessName()
+        {
+            string commandName = "sleep";
+
+            // sleep program doesn't exist on some distros
+            if (IsProgramInstalled(commandName))
+            {
+                string tmpDir = Path.Combine(TestDirectory, GetTestFileName());
```

Comment:
I think you can put it in TestDirectory directly?

Reply:
Then you wouldn't need to delete the directory

Reply:
good idea